net/HttpRequest.hpp

```diff
@@ -1609,11 +1614,35 @@ class Session final : public ProtocolHandlerInterface
 
             if (!socket->send(_request))
             {
+                _result = net::asyncConnectResult::SocketError;
                 LOG_ERR("Error while writing to socket");
             }
         }
     }
 
+    void callOnConnectFail()
+    {
+        if (!_onConnectFail)
+            return;
+
+        auto self = shared_from_this();
```

Comment:
Nit: https://github.com/CollaboraOnline/online/blob/cb100b9c4a5a8ab93e0edbcbb624b0b68cf2fb23/README.CONTRIBUTING.md#c encourages conservative use of 'auto', you could spell out the type here, so...